dcpy/connectors/edm/packaging.py

```diff
@@ -0,0 +1,150 @@
+import os
+import shutil
+from pathlib import Path
+from typing import Callable
+from dataclasses import dataclass
+import typer
+
+from dcpy.utils import s3
+from dcpy.utils.logging import logger
+from dcpy.connectors.edm import publishing
+
+BUCKET = "edm-distributions"
+BUCKET_ACL = s3.string_as_acl("public-read")
+
+DOWNLOAD_ROOT_PATH = Path(".library")
+OUTPUT_ROOT_PATH = Path(".output")
+
+
+@dataclass
+class PackageKey(publishing.ProductKey):
+    product: str
+    version: str
+
+    def __str__(self):
+        return f"{self.product} - {self.version}"
+
+    @property
+    def path(self) -> str:
+        return f"{self.product}/{self.version}"
+
+
+def colp(publish_key: publishing.PublishKey, package_key: PackageKey):
+    raise NotImplementedError
+
+
+def templatedb(publish_key: publishing.PublishKey, package_key: PackageKey):
```

Comment:
Just for my own understanding - seems that what we're going for with one of these functions is
- download our "official" output in edm-publishing/{dataset}/{version}
- "package" it locally (in the new .output folder). So .library -> .output
- push the packaged version to edm-distributions (not in this call, but logically the next step)

Reply:
yup! noting though that, like the push, the download isn't in this call either and it's from edm-publishing/{dataset}/**publish**/{version}

and might be nice to have these functions be more structured/class-based, but I didn't wanna invest too much into making this part rigid when the actual packaging transformations are gonna vary so much by product and haven't really been attempted yet